src/sentry/sentry_metrics/indexer/models.py

```diff
@@ -50,3 +56,76 @@ class Meta:
         constraints = [
             models.UniqueConstraint(fields=["string", "organization_id"], name="unique_org_string"),
         ]
+
+
+class StringIndexerCache:
+    def __init__(self, model: Any, version: int):
+        self.model = model
+        self.version = version
+        self.cache_ttl: int = settings.SENTRY_METRICS_INDEXER_CACHE_TTL
+
+    @property
+    def randomized_ttl(self) -> int:
+        # introduce jitter in the cache_ttl so that when we have large
+        # amount of new keys written into the cache, they don't expire all at once
+        jitter = random.uniform(0, 0.25) * self.cache_ttl
+        return int(self.cache_ttl + jitter)
+
+    def make_cache_key(self, key: Union[str, int]) -> str:
```

Comment:
This is a stylistic thing, but you could made a custom type `Key = Union[str, int]` and then use that throughout. I think it would make the code a little easier to read. If you make some of the functions generic things might get a little weird though.